extraction/clip_shape.py

```diff
@@ -0,0 +1,47 @@
+import glob
+import os
+import geopandas as gpd
+from osgeo import gdal
+
+RAW_DATA_DIR = os.path.join("..", "data", "raw")
+CONVERTED_DATA_DIR = os.path.join("..", "data", "converted")
+
+shapefile = gpd.read_file(
+    os.path.join(RAW_DATA_DIR, "overlay", "ST_PV_Potenzial_2013.zip")
+)
+
+raster_dir = os.path.join(CONVERTED_DATA_DIR, "raster")
+
+out_dir = os.path.join(CONVERTED_DATA_DIR, "vector")
+if not os.path.exists(out_dir):
+    os.makedirs(out_dir)
+
+raster_filenames = glob.glob(os.path.join(raster_dir, "*.tif"))
+
+for raster_file in raster_filenames:
+    print(f"Processing {raster_file}")
+
+    # get the extent of the current raster
+    ds = gdal.Open(raster_file)
+    x_min, x_max, y_min, y_max = ds.GetGeoTransform()[0], ds.GetGeoTransform()[0] + ds.RasterXSize * ds.GetGeoTransform()[
+        1], ds.GetGeoTransform()[3] + ds.RasterYSize * ds.GetGeoTransform()[5], ds.GetGeoTransform()[3]
+    ds = None
```

Comment:
I think you want to use
`del ds`
here?

Reply:
yes, more pythonic but gdal code generally follows the other way so will leave for now